lib/linked_list.rb

```diff
@@ -18,129 +19,255 @@ def initialize
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity:  
-    # Space Complexity
+    # Time Complexity: O(1) (constant), because the length of the list doesn't matter
+    # Space Complexity: O(n), where n is the length of the list

```

Comment:
But the method itself doesn't add more space to the list, it only adds 1 element.